engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -391,58 +398,43 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     runTransaction(action)
   }
 
+  private def futureDescriptorFromExecutionAndAux(executionAndAux: ExecutionAndAux)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
+    val sources = WorkflowSourceFiles(executionAndAux.aux.wdlSource.toRawString, executionAndAux.aux.jsonInputs.toRawString, executionAndAux.aux.workflowOptions.toRawString)
+    materializeWorkflowDescriptor(WorkflowId(UUID.fromString(executionAndAux.execution.workflowExecutionUuid)), sources)
+  }
+
   override def getWorkflow(workflowExecutionId: Int)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
     val action = for {
       workflowExecutionResult <- dataAccess.workflowExecutionsByPrimaryKey(workflowExecutionId).result.head
       workflowAux <- dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
-      workflowDescriptor = WorkflowDescriptor(
-        WorkflowId(UUID.fromString(workflowExecutionResult.workflowExecutionUuid)),
-        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
-      )
-    } yield workflowDescriptor
+    } yield ExecutionAndAux(workflowExecutionResult, workflowAux)
 
-    runTransaction(action)
+    runTransaction(action) flatMap futureDescriptorFromExecutionAndAux
   }
 
   override def getWorkflow(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
     val action = for {
       workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
       workflowAux <- dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
-      workflowDescriptor = WorkflowDescriptor(
-        workflowId,
-        WorkflowSourceFiles(workflowAux.wdlSource.toRawString, workflowAux.jsonInputs.toRawString, workflowAux.workflowOptions.toRawString)
-      )
-    } yield workflowDescriptor
+    } yield ExecutionAndAux(workflowExecutionResult, workflowAux)
 
-    runTransaction(action)
+    runTransaction(action) flatMap futureDescriptorFromExecutionAndAux
   }
 
   override def getWorkflowsByState(states: Traversable[WorkflowState])
                                   (implicit ec: ExecutionContext): Future[Traversable[WorkflowDescriptor]] = {
     val action = for {
       workflowExecutionResults <- dataAccess.workflowExecutionsByStatuses(states.map(_.toString)).result
+      workflowAuxSeq <- DBIO.sequence(workflowExecutionResults.map( workflowExecutionResult =>
+        dataAccess.workflowExecutionAuxesByWorkflowExecutionUuid(workflowExecutionResult.workflowExecutionUuid).result.head
+      ))
+      executionAndAux <- DBIO.sequence((workflowExecutionResults zip workflowAuxSeq) map { case (exec, aux) => DBIO.successful(ExecutionAndAux(exec, aux)) })
+    } yield executionAndAux
```

Comment:
But doing the zip/map in the yield would allow you to get rid of the `DBIO.sequence` and `DBIO.successful`.

```scala
    } yield (workflowExecutionResults zip workflowAuxSeq) map ExecutionAndAux.tupled
```